Test/ImportDelamDataTest.cpp

```diff
@@ -0,0 +1,621 @@
+ * other materials provided with the distribution.
+ *
+ * Neither the name of BlueQuartz Software, the US Air Force, nor the names of its
+ * contributors may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+ * USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The code contained herein was partially funded by the followig contracts:
+ *    United States Air Force Prime Contract FA8650-15-D-5231
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ */
+
+#pragma once
+
+#include <fstream>
+
+#include "SIMPLib/DataContainers/DataContainerArray.h"
+#include "SIMPLib/Filtering/FilterManager.h"
+#include "SIMPLib/Geometry/RectGridGeom.h"
+
+#include "UnitTestSupport.hpp"
+
+#include "SimulationIO/SimulationIOFilters/ImportDelamData.h"
+
+#include "SimulationIOTestFileLocations.h"
+
+class ImportDelamDataTest
+{
+  const std::string k_CSDGMTestFile = UnitTest::TestTempDir.toStdString() + "/csdgm.txt";
+  const std::string k_BvidStdOutTestFile = UnitTest::TestTempDir.toStdString() + "/bvid_stdout.txt";
+  const std::string k_WrongCSDGMTestFile = UnitTest::TestTempDir.toStdString() + "/csdgm.exe";
+  const std::string k_WrongBvidStdOutTestFile = UnitTest::TestTempDir.toStdString() + "/bvid_stdout.rar";
+  const char k_NL = '\n';
+
+public:
+  ImportDelamDataTest() = default;
+  ~ImportDelamDataTest() = default;
+  ImportDelamDataTest(const ImportDelamDataTest&) = delete;            // Copy Constructor
+  ImportDelamDataTest(ImportDelamDataTest&&) = delete;                 // Move Constructor
+  ImportDelamDataTest& operator=(const ImportDelamDataTest&) = delete; // Copy Assignment
+  ImportDelamDataTest& operator=(ImportDelamDataTest&&) = delete;      // Move Assignment
+
+  // -----------------------------------------------------------------------------
+  int TestFilterAvailability()
+  {
+    // Now instantiate the ImportDelamDataTest Filter from the FilterManager
+    QString filtName = "ImportDelamData";
+    FilterManager* fm = FilterManager::Instance();
+    IFilterFactory::Pointer filterFactory = fm->getFactoryFromClassName(filtName);
+    if (nullptr == filterFactory.get())
```

Comment:
```suggestion
    if (nullptr == filterFactory)
```
You can compare smart pointers directly with nullptr